### PR TITLE
Feature/#7/token picker

### PR DESCRIPTION
**Pull request by alfetopito**
# Update Sep 1

- Loading token balance for safe when selecting a token
- Loading token images
- Displaying fallback img when no img is available
- Storybook fixes

screenshot_2020-09-01_08-55-05 screenshot_2020-09-01_08-54-55

----

# Description

~:warning: WIP :warning:~

Token selector component defined on [the design](https://projects.invisionapp.com/d/main/default/#/console/20213446/427462294/preview)
screenshot_2020-08-27_15-19-52

- Fetching token balance for given safe address on select
- NOT yet able to test it on Storybook. Can't tell why but the story won't show up :/
- It does work if I try it on Rinkeby, though (don't mind the style, still in progress):
screenshot_2020-08-27_15-10-43

# To Test
- [ ] Style should match design
- [ ] Should be able to pick token from the list and see balance for it

# Background

The issues I'm facing on the Storybook are very likely due to how the component is implemented.
There are hooks withing it, and I'm trying to find ways to make it work with Storybook without workarounds.

I've read a few approaches to how to test hooks on Storybook:
1. https://www.reddit.com/r/reactjs/comments/drxj8p/mocking_hooks_inside_stories/
2. https://www.npmjs.com/package/react-hooks-compose
3. https://stackoverflow.com/questions/56497565/how-to-mock-a-custom-hook-inside-of-a-react-component-you-want-to-test
4. https://dev.to/emma/how-to-mock-react-hooks-using-dependency-injection-1lje
5. https://storybook.js.org/docs/react/workflows/build-pages-with-storybook#avoiding-mocking-dependencies

This code is an attempt at `5`, having a local context provider, although I'm not convinced it's making it any better than having 2 components like the other approaches suggest in a sort of dependency injection way.

Still, not able to make the story show up on the Storybook at all.

Ideas on what's missing?

### CODE REVIEW

File: src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      iconUrl: "",
+    }, // gno
+    {
+      id: "0x1A5F9352Af8aF974bFC03399e3767DF6370d82e4",
+      label: "OWL",
+      subLabel: "",
+      iconUrl: "",
+    }, // owl
+  ],
+  tokenBalance: new BN(10000000000000000000000),
```
Comment (Velenir) [inline src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx:44]:
```suggestion
  tokenBalance: new BN('10000000000000000000000'),
```

lol

File: src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    name: "",
+    symbol: "OWL",
+    decimals: 18,
+  },
+};
+
+export default {
+  component: TokenSelectorContainer,
+  title: "basic/input/TokenSelector",
+  excludeStories: /.*Data$/,
```
Comment (Velenir) [inline src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx:56]: btw, `/.*Data$/` same as `/Data$/`

**Review by Velenir — APPROVED** [review]

File: src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx (hunk 2 above)
Comment (alfetopito) [reply src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx:56]: True. I just got this from storybook docs and didn't bother checking.

File: src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx (hunk 1 above)
Comment (alfetopito) [reply src/components/basic/inputs/TokenSelector/TokenSelector.stories.tsx:44]: You gotta be fucking kidding :facepalm:

File: src/hooks/useTokenBalance.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const [balance, setBalance] = useState<BN | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { safeInfo, getContract } = useContext(Web3Context);
+
+  // TODO: cache responses so we don't have to query the network on every single change
+  // TODO: also, we should subscribe to new block header if available and query the balance
+  // on every change
+  useEffect(() => {
+    setIsLoading(true);
```
Comment (andre-meyer) [inline src/hooks/useTokenBalance.ts:27]: Could the below code fail? If so, it makes sense to alter this to have a status for `ERROR` aswell.

**Review by andre-meyer — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -17,6 +17,7 @@
   "dependencies": {
     "@gnosis.pm/dex-contracts": "^0.4.1",
     "@gnosis.pm/dex-liquidity-provision": "gnosis/dex-liquidity-provision#cmm-changes",
+    "@gnosis.pm/dex-js": "^0.4.1",
```
Comment (anxolin) [inline package.json:20]: hehe, nice u make use of it!

Comment (alfetopito) [reply package.json:20]: Oh yes, and I intend at some point to add more stuff to it.
Would you be willing to have more contracts added to `dex-js`? :nerd_face:

File: src/hooks/useTokenBalance.ts (hunk 3 above)
Comment (alfetopito) [reply src/hooks/useTokenBalance.ts:29]: https://github.com/gnosis/safe-cmm-app-react/pull/24/commits/a1f7bc2a1fb6ba4cfa9a039f9db13a3528fbe7b7

File: package.json (hunk 4 above)
Comment (anxolin) [reply package.json:20]: It doesn't bother me if they are GP related why not :) 
we don't need to import them in the projects, so as long as they are tree-shakeable is good

**Pull request merged**